0019/README.md

```diff
@@ -0,0 +1,157 @@
+---
+name: Endpoint Metadata Management Protocol
+shortname: 19/EPMMP
+status: raw
+editor: Vitalii Kozlovskyi <vkozlovskyi@kaaiot.io>
+---
+
+
+<!-- toc -->
+
+
+# Introduction
+
+Endpoint Metadata Management Protocol (EPMMP) is designed for managing endpoint metadata between Kaa services.
+
+EPMMP complies with the [Inter-Service Messaging](/0003/README.md) guidelines.
+
+
+# Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+The following terms and definitions are used in this RFC.
+
+- **Endpoint metadata repository (repository)**: any service that exposes the EPMMP interface to other services for managing endpoint metadata.
+
+- **Endpoint metadata management client (client)**: any service that uses exposed EPMMP interface to manage endpoint metadata.
+
+
+# Design
+
+## Endpoint metadata get request
+
+*Endpoint metadata get request* message is a [targeted message](/0003/README.md#targeted-messaging) that the client sends to the repository to retrieve endpoint metadata.
+
+The client MUST send endpoint metadata get request messages using the following NATS subject:
+```
+kaa.v1.service.{repository-service-instance-name}.epmmp.ep-metadata-get-request
+```
+where:
+- `{repository-service-instance-name}` is the endpoint metadata repository service instance name.
+
+The client MUST include NATS `replyTo` field to handle the response.
+It is RECOMMENDED to follow the subject format described in [3/ISM session affinity section](/0003/README.md#session-affinity):
+```
+kaa.v1.replica.{client-service-replica-id}.epmmp.ep-metadata-get-response
+```
```

Comment:
It should be a unique subject because we will match the response by its. Cover this moment here, plz

Reply:
It is a recommendation. It was done to keep consistency with other RFCs.

Reply:
if would you follow this recommendation you will never receive a response. Really good one)

Reply:
It is RFC, not actual implementation.

Reply:
And I think if there no consistency in code, the same should be in docs. Better when you don't have docs, than have mislead one

Reply:
You confuse the RFC with the implementation documentation.